handlers/search.js

```diff
@@ -91,28 +95,36 @@ function searchPage(request, reply) {
   const filters = getFilters(query);
   const advancedSearch = query.advanced_search || (Object.keys(filters).length > 0);
 
-  trials.search(queryStr, page, perPage, filters).then((_trials) => {
-    const currentPage = page || 1;
-    const pagination = generatePaginationLinks(
-      request.url, currentPage,
-      perPage, maxPages,
-      _trials.total_count
-    );
-    const queryDescription = queryStr || 'clinical trial data';
-    const title = `Search results for ${queryDescription} in OpenTrials`;
-    const description = `OpenTrials has ${_trials.total_count} matches for
+  sources.list().then((_sources) => {
+    trials.search(queryStr, page, perPage, filters).then((_trials) => {
```

Comment:
You have a bug here. As you're not returning the inner promise, JavaScript doesn't guarantee that it'll be ran before the rest of the code. You need to return it, so all promises are part of the same chain and JS guarantees that they'll run in sequence. The styling that could be improved as well. I think the best would be:

```javascript
Promise.join(
  sources.list(),
  trials.search(queryStr, page, perPage, filters),
  (_sources, _trials) => {
    // ...
  }
).catch((err) => {
  // ...
})
```